mmpretrain/models/backbones/hivit.py

```diff
@@ -0,0 +1,609 @@
+                Defaults to ``dict(type='LN')``.
+            kernel_size (int): Kernel size.
+            pad_size (int): Pad size.
+    """
+    def __init__(
+            self, img_size=224, patch_size=16, inner_patches=4, in_chans=3, embed_dim=128,
+            norm_cfg=None, kernel_size=None, pad_size=None):
+        super().__init__()
+        img_size = to_2tuple(img_size) if not isinstance(img_size, tuple) else img_size
+        patch_size = to_2tuple(patch_size)
+        patches_resolution = [img_size[0] // patch_size[0], img_size[1] // patch_size[1]]
+        self.img_size = img_size
+        self.patch_size = patch_size
+        self.inner_patches = inner_patches
+        self.patches_resolution = patches_resolution
+        self.num_patches = patches_resolution[0] * patches_resolution[1]
+
+        self.in_chans = in_chans
+        self.embed_dim = embed_dim
+
+        conv_size = [size // inner_patches for size in patch_size]
+        kernel_size = kernel_size or conv_size
+        pad_size = pad_size or 0
+        self.proj = nn.Conv2d(in_chans, embed_dim, kernel_size=kernel_size, stride=conv_size, padding=pad_size)
+        if norm_cfg is not None:
+            self.norm = build_norm_layer(norm_cfg, embed_dim)
+        else:
+            self.norm = None
+
+    def forward(self, x):
+        B, C, H, W = x.shape
+        patches_resolution = (H // self.patch_size[0], W // self.patch_size[1])
+        num_patches = patches_resolution[0] * patches_resolution[1]
+        x = self.proj(x).view(
+            B, -1,
+            patches_resolution[0], self.inner_patches,
+            patches_resolution[1], self.inner_patches,
+        ).permute(0, 2, 4, 3, 5, 1).reshape(B, num_patches, self.inner_patches, self.inner_patches, -1)
+        if self.norm is not None:
+            x = self.norm(x)
+        return x
+
+
+class PatchMerge(nn.Module):
+    """
+        PatchMerge
+
+        Args:
+            dim (int): Number of input channels.
+            norm_cfg (dict): Config dict for normalization layer.
+    """
+    def __init__(self, dim, norm_cfg):
+        super().__init__()
+        self.norm = build_norm_layer(norm_cfg, dim * 4)
+        self.reduction = nn.Linear(dim * 4, dim * 2, bias=False)
+
+    def forward(self, x, *args, **kwargs):
+        is_main_stage = len(x.shape) == 3
+        if is_main_stage:
+            B, N, C = x.shape
+            S = int(math.sqrt(N))
+            x = x.reshape(B, S // 2, 2, S // 2, 2, C) \
+                .permute(0, 1, 3, 2, 4, 5) \
+                .reshape(B, -1, 2, 2, C)
+        x0 = x[..., 0::2, 0::2, :]
+        x1 = x[..., 1::2, 0::2, :]
+        x2 = x[..., 0::2, 1::2, :]
+        x3 = x[..., 1::2, 1::2, :]
+
+        x = torch.cat([x0, x1, x2, x3], dim=-1)
+        x = self.norm(x)
+        x = self.reduction(x)
+
+        if is_main_stage:
+            x = x[:, :, 0, 0, :]
+        return x
+
+
+@MODELS.register_module()
+class HiViT(nn.Module):
```

Comment:
please inherit from `BaseBackbone` and add `init_cfg` to function `__init__` args, you could refer to code of swin